ethcore/engines/authority-round/src/lib.rs

```diff
@@ -1277,22 +1275,20 @@ impl Engine for AuthorityRound {
 			.map(ToString::to_string)
 			.unwrap_or_default();
 
-		let mut info = map![
-			"step".into() => step,
-			"signature".into() => signature
-		];
+		let mut info = BTreeMap::new();
+		info.insert("step".into(), step);
+		info.insert("signature".into(), signature);
 
 		if header.number() >= self.empty_steps_transition {
-			let empty_steps =
-				if let Ok(empty_steps) = header_empty_steps(header).as_ref() {
-					format!("[{}]",
-						empty_steps.iter().fold(
-							"".to_string(),
-							|acc, e| if acc.len() > 0 { acc + ","} else { acc } + &e.to_string()))
-
-				} else {
-					"".into()
-				};
+			let empty_steps = header_empty_steps(header).as_ref().map_or(String::new(), |empty_steps| {
```

Comment:
this change is also unrelated, I found this piece very hard to read especially the `fold`